src/test/utxocommit_tests.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2018 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+// Tests for CUtxoCommit wrapper.
+// Mostly redundant with libsecp256k1_multiset tests
+
+#include "coins.h"
+#include "test/test_bitcoin.h"
+#include "test/testutil.h"
+#include "util.h"
+#include "random.h"
+
+#include <vector>
+
+#include <boost/test/unit_test.hpp>
+
+#include "secp256k1/include/secp256k1_multiset.h"
+#include "utxocommit.h"
+
+static COutPoint RandomOutpoint() {
+    const COutPoint op(GetRandHash(), (uint32_t)GetRandInt(1000));
+    return op;
+}
+
+static Coin RandomCoin() {
+    const int scriptlen = GetRandInt(0x3f);
+    auto script = std::vector<unsigned char>(scriptlen);
+    GetRandBytes(&script[0], scriptlen);
+    const Coin c(CTxOut(CAmount(GetRandInt(1000)), CScript(script)),
+                 GetRandInt(10000), GetRandInt(1));
+    return c;
+}
+
+BOOST_FIXTURE_TEST_SUITE(utxocommit_tests, TestingSetup)
+
+BOOST_AUTO_TEST_CASE(utxo_commit_order) {
+
+    // Test order independence
+
+    const COutPoint op1 = RandomOutpoint();
+    const COutPoint op2 = RandomOutpoint();
+    const COutPoint op3 = RandomOutpoint();
+    const Coin c1 = RandomCoin();
+    const Coin c2 = RandomCoin();
+    const Coin c3 = RandomCoin();
+
+    CUtxoCommit uc1, uc2, uc3;
+    BOOST_CHECK(uc1 == uc2);
+    uc1.Add(op1, c1);
+    uc1.Add(op2, c2);
+    uc1.Add(op3, c3);
+
+    uc2.Add(op2, c2);
+    BOOST_CHECK(uc1 != uc2);
+    uc2.Add(op3, c3);
+    uc2.Add(op1, c1);
+    BOOST_CHECK(uc1 == uc2);
+
+    // remove ordering
+    uc2.Remove(op2, c2);
+    uc2.Remove(op3, c3);
+
+    uc1.Remove(op2, c2);
+    uc1.Remove(op3, c3);
```

Comment:
Did you mean to reverse the order of these removes?

Reply:
Yes. Fixed now.